src/main/java/mate/academy/bookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,143 @@
+package mate.academy.bookshop.service.impl;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.order.OrderItemResponseDto;
+import mate.academy.bookshop.dto.order.OrderRequestDto;
+import mate.academy.bookshop.dto.order.OrderResponseDto;
+import mate.academy.bookshop.dto.status.StatusDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.OrderItemMapper;
+import mate.academy.bookshop.mapper.OrderMapper;
+import mate.academy.bookshop.model.CartItem;
+import mate.academy.bookshop.model.Order;
+import mate.academy.bookshop.model.OrderItem;
+import mate.academy.bookshop.model.ShoppingCart;
+import mate.academy.bookshop.model.Status;
+import mate.academy.bookshop.model.User;
+import mate.academy.bookshop.repository.OrderItemRepository;
+import mate.academy.bookshop.repository.OrderRepository;
+import mate.academy.bookshop.repository.OrderStatusRepository;
+import mate.academy.bookshop.repository.ShoppingCartRepository;
+import mate.academy.bookshop.repository.UserRepository;
+import mate.academy.bookshop.service.OrderService;
+import mate.academy.bookshop.service.ShoppingCartService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private static final Status.StatusName DEFAULT_STATUS_NAME
+            = Status.StatusName.PENDING;
+    private final ShoppingCartService shoppingCartService;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderStatusRepository statusRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderResponseDto createOrder(Long userId, OrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = getCartById(userId);
+        Order newOrder = new Order();
+        newOrder.setUser(getUserById(userId));
+        newOrder.setTotal(getTotal(shoppingCart.getCartItems()));
+        newOrder.setShippingAddress(requestDto.getShippingAddress());
+        newOrder.setTotal(getTotal(shoppingCart.getCartItems()));
+        newOrder.setOrderDate(LocalDateTime.now());
+        newOrder.setStatus(getByName(DEFAULT_STATUS_NAME));
+        orderRepository.save(newOrder);
+        Set<OrderItem> orderItems
+                = mapAndSaveOrderItems(shoppingCart.getCartItems(), newOrder);

```

Comment:
OPTIONAL: It is possible to use cascades and avoid this db request to save order items